### DISCUSSION TITLE
Wrong machine coordinates after hitting the Stop button?

### DISCUSSION THREAD

**Discussion by the-powl** [winder/Universal-G-Code-Sender#2046] [Help]
Hi,

I'm using grbl 1.1. I've had it several times now that when I abort a milling job by pressing the Stop button in UGS the machine coordinates will be wrong after that. The zero position (G0X0Y0Z0) is then wrong up to a few centimeters. Actually I thought when I stop the current job the machine safely comes to a halt and the sender just stops sending new gcode.

Any Ideas about what's going wrong here? Or is this just normal behaviour?

**Comment by breiler** [comment]
No this is not normal. When hitting stop it should pause the machine to bring it to a halt. Once it reaches a HOLD-state it will reset the controller to clear all buffers but should retain the correct position.

If the position is lost it could mean that it doesn't have the time to decelerate the machine to a hold before doing the reset. I find this a bit unlikely though as we are waiting a total of 50 x 200ms ~ 10 seconds for it to halt. Other than that I don't know... 

What is your settings (type $$ in the console)? What is your GRBL version (type $I in the console).
